crates/polars-mem-engine/src/scan_predicate/functions.rs

```diff
@@ -431,6 +438,9 @@ where
     *row_count = None;
 
     if selected_path_indices.clone().next() != Some(0) {
```

Comment:
We rather have estimations than `None`. So I think we should multiply the estimation with the ratio of `total_files/ remaining_files`.